FinalTask/FinalTests.cs

```diff
@@ -0,0 +1,220 @@
+        {
+            var authenticationPage = mainPage.ToSignInPage();
+            var myAccountpage = authenticationPage.SignIn("123@test.test", "123456");
+            var wishListPage = myAccountpage.ToMyWishListPage();
+
+            var productDetailsPage = wishListPage.ToProductDetailsPage();
+            productDetailsPage.AddToWishList();
+            wishListPage.Load();
+
+            Assert.True(!wishListPage.IsEmptyWishList(), "Wishlist is empty");
+        }
+
+        [Test]
+        [AllureSubSuite("Final task")]
+        [AllureSeverity(SeverityLevel.Critical)]
+        [AllureLink("http://automationpractice.com/")]
+        [AllureTest("Own wish list test")]
+        [AllureOwner("Alexandra Egorova")]
+        public void OwnWishList()
+        {
+            var authenticationPage = mainPage.ToSignInPage();
+            var myAccountpage = authenticationPage.SignIn("123@test.test", "123456");
+            var wishListPage = myAccountpage.ToMyWishListPage();
+            wishListPage.AddWishList("wishlist");
+
+            var productDetailsPage = wishListPage.ToProductDetailsPage();
+            productDetailsPage.AddToWishList();
+            wishListPage.Load();
+
+            Assert.True(wishListPage.WishListItemsCount("wishlist") == "1", "Product was not added to a wishlist");
+        }
+
+        [Test]
+        [AllureSubSuite("Final task")]
+        [AllureSeverity(SeverityLevel.Critical)]
+        [AllureLink("http://automationpractice.com/")]
+        [AllureTest("Add to cart test")]
+        [AllureOwner("Alexandra Egorova")]
+        public void Cart()
+        {
+            var authenticationPage = mainPage.ToSignInPage();
+            authenticationPage.SignIn("123@test.test", "123456");
+            mainPage.Load();
+            var cartPage = mainPage.ToShoppingCart();
+            cartPage.CleanCart();
+            mainPage.Load();
+            var selectedItems = mainPage.AddItemsToCard(3);
+            cartPage = mainPage.ToShoppingCart();
+            var itemsInCart = cartPage.GetItemsInCart();
+
+            Assert.True(selectedItems.Length == itemsInCart.Count, "Not all items were added to cart");
+
+            double selectedItemsTotal = 0;
+            foreach (var item in selectedItems)
+            {
+                Assert.True(itemsInCart.Contains(item.Name), "Item was not added to cart");
+                selectedItemsTotal = selectedItemsTotal + item.Price;
+            }
+
+            var itemsInCartTotal = cartPage.GetTotal();
+
+            Assert.True(selectedItemsTotal == itemsInCartTotal, "Totals do not match");
+        }
+
+        [TearDown]
+        public void HandleError()
+        {
+            if (TestContext.CurrentContext.Result.Outcome.Status == TestStatus.Failed)
+            {
+                var image = ((ITakesScreenshot)driver).GetScreenshot();
+                image.SaveAsFile(@"C:\Users\aleks\Downloads\Failed.png", ScreenshotImageFormat.Png);
```

Comment:
It's not recommended to use dependency path in code. Your HandleError method will fail in case I run your tests on my local machine. Just put images and other any output files from your code in solution root.  Use Environment.CurrentDirectory or whatever you want (there are several ways how to get project base location)